src/routes/RoutePage.jsx

```diff
@@ -7,24 +13,69 @@ import {
   LikedVideos,
   Page404,
   Playlist,
+  VideoDetail,
   WatchLater,
 } from "../pages/pages-index";
+import axios from "axios";
+import { useDataContext } from "../context/useContext-index";
+import { ScrollToTop } from "../services/ScrollToTop";
+
+const VIDEOS_API = "/api/videos";
 
 function RoutePage() {
+  const { dataState, dataDispatch } = useDataContext();
+  const [loading, setLoading] = useState(false);
+  let { categoryName, videoID, all } = useParams;
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const fetchData = async () => {
+    try {
+      const videos = await axios(VIDEOS_API);
+      const categories = await axios("/api/categories");
+      dataDispatch({ type: "SET_VIDEOS", payload: videos.data.videos });
+      dataDispatch({
+        type: "SET_CATEGORIES",
+        payload: categories.data.categories,
+      });
+
+      setLoading(true);
+    } catch (err) {
+      console.log(err);
+    }
+  };
```

Comment:
Suggestion: move this function to a separate file

Reply:
will do make common function for all API calls later